src/www/ui/api/index.php

```diff
@@ -246,11 +246,11 @@ function (\Slim\Routing\RouteCollectorProxy $app) {
 ////////////////////////////ADMIN-USERS/////////////////////
 $app->group('/users',
   function (\Slim\Routing\RouteCollectorProxy $app) {
-    $app->get('[/{id:\\d+}]', UserController::class . ':getUsers');
-    $app->put('[/{id:\\d+}]', UserController::class . ':updateUser');
-    $app->post('', UserController::class . ':addUser');
-    $app->delete('/{id:\\d+}', UserController::class . ':deleteUser');
     $app->get('/self', UserController::class . ':getCurrentUser');
+    $app->get('[/{pathParam:[\\w\\-\\.]+}]', UserController::class . ':getUsers');
+    $app->put('[/{pathParam:[\\w\\-\\.]+}]', UserController::class . ':updateUser');
```

Comment:
However, this regex will not work for API V1 which expects the path variable to be a digit. Also, will fail for organizations which uses email id as username (does not allow `@`).

How about using `[\w\d\-\.@_]+` as regex? It will allow every word, digit and other special characters commonly used in emails (`-.@_`). Please feel free to expand and experiment.
Also, once you've settled on the pattern, put it in a variable as it is used at multiple locations and keeping track to update pattern in future will be difficult.

Reply:
Hm..You are right. Also I think `[\w\d\-\.@_]+` looks good enough. I was searching for some more additions but those are very uncommon and might not be useful for our usecase. Although I need thoughts on finalising this. Additions I found were adding ``!#$%&'*+-/=?^_`{|}~`` and `"(),:;<>@[\]` characters for local part of the email. Domain part is already covered by the expression you suggested. WDYT?

Reply:
Lets keep it simple and not do a complete validation.